File: 0071-simplify-path/0071-simplify-path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        path = path.split("/");
        
        stack = []
        
        for string in path:
            if stack and string == "..":
                stack.pop()
            elif not (len(string) < 1 or string == "." or string == ".."):
                stack.append(string)
        
        return "/%s"%('/'.join(stack))

File: 0071-simplify-path/0071-simplify-path.js
/**
 * @param {string} path
 * @return {string}
 */
var simplifyPath = function(path) {
    path = path.split("/");
    let arr = [];
    
    for (let str of path){
        if (str == ".."){
            arr.pop();
        } else if (!(str.length < 1 || str == ".")){
            arr.push(str);
        }
    }
    
    return `/${arr.join("/")}`;
};